Jellyfin.Api/Helpers/MediaInfoHelper.cs

```diff
@@ -255,10 +255,27 @@ public class MediaInfoHelper
                 streamInfo.PlaySessionId = playSessionId;
                 streamInfo.StartPositionTicks = startTimeTicks;
 
-                mediaSource.SupportsDirectPlay = streamInfo.PlayMethod == PlayMethod.DirectPlay;
+                if (mediaSource.SupportsDirectPlay)
+                {
+                    mediaSource.SupportsDirectPlay = streamInfo.PlayMethod == PlayMethod.DirectPlay;
+                }
+
                 // Players do not handle this being set according to PlayMethod
-                mediaSource.SupportsDirectStream = options.EnableDirectStream ? streamInfo.PlayMethod == PlayMethod.DirectPlay || streamInfo.PlayMethod == PlayMethod.DirectStream : streamInfo.PlayMethod == PlayMethod.DirectPlay;
-                mediaSource.SupportsTranscoding = streamInfo.PlayMethod == PlayMethod.DirectStream || mediaSource.TranscodingContainer != null;
+                if (mediaSource.SupportsDirectStream)
+                {
+                    mediaSource.SupportsDirectStream =
+                        options.EnableDirectStream
+                            ? streamInfo.PlayMethod == PlayMethod.DirectPlay || streamInfo.PlayMethod == PlayMethod.DirectStream
+                            : streamInfo.PlayMethod == PlayMethod.DirectPlay;
+                }
+
+                if (mediaSource.SupportsTranscoding)
+                {
+                    mediaSource.SupportsTranscoding =
+                        streamInfo.PlayMethod == PlayMethod.DirectStream
+                        || mediaSource.TranscodingContainer != null
+                        || profile.TranscodingProfiles.Any(i => i.Type == streamInfo.MediaType && i.Context == options.Context);
+                }
```

Comment:
Remove `if` - doesn't work for remuxing.
Since I am not sure what to do with these `Supports*`, let's remove all 'if's.

Reply:
```suggestion
                mediaSource.SupportsTranscoding =
                    streamInfo.PlayMethod == PlayMethod.DirectStream
                    || mediaSource.TranscodingContainer != null
                    || profile.TranscodingProfiles.Any(i => i.Type == streamInfo.MediaType && i.Context == options.Context);
```